be/src/storage/meta_reader.cpp

```diff
@@ -125,21 +127,22 @@ Status MetaReader::_get_segments(const TabletSharedPtr& tablet, const Version& v
         return Status::OK();
     }
 
-    std::vector<RowsetSharedPtr> rowsets;
     Status acquire_rowset_st;
     {
         std::shared_lock l(tablet->get_header_lock());
-        acquire_rowset_st = tablet->capture_consistent_rowsets(_version, &rowsets);
+        acquire_rowset_st = tablet->capture_consistent_rowsets(_version, &_rowsets);
```

Comment:
Nit:
you can capture rowsets with a local variable. After rowsets was captured successfully.
Then you can assign 
```
_rowsets = std::move(rowsets);
```
Then, you don't  need to clear _rowsets when process will return ahead.